layers/layers_windows_test.go

```diff
@@ -0,0 +1,45 @@
+package layers_test
+
+import (
+	"archive/tar"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	h "github.com/buildpacks/lifecycle/testhelpers"
+)
+
+func parentHeader(filePath string, fi os.FileInfo) *tar.Header {
+	hdr := &tar.Header{
+		Typeflag: tar.TypeDir,
+	}
+	hdr.Name = tarPath(filePath)
+	return hdr
+}
+
+func tarPath(filePath string) string {
+	return path.Join("Files", filepath.ToSlash(strings.TrimPrefix(
+		filePath,
+		filepath.VolumeName(filePath)+`\`,
+	)))
+}
+
+func assertOSSpecificFields(t *testing.T, expected *tar.Header, hdr *tar.Header) {
+	t.Helper()
+	h.AssertEq(t, hdr.Format, tar.FormatPAX)
+}
+
+func assertOSSpecificEntries(t *testing.T, tr *tar.Reader) {
+	for _, windowsEntry := range []string{"Files", "Hives"} {
```

Comment:
This seems to be checking that the first two entries are of type directory, but don't we want to add some sort of assertion around `header.Name == windowsEntry`?

Reply:
Good point!